turbinia/processors/resource_manager_test.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for resource_manager handler."""
+
+from __future__ import unicode_literals
+
+import unittest
+import os
+import shutil
+import tempfile
+
+import mock
+
+from turbinia import TurbiniaException, config
+from turbinia.processors import resource_manager
+
+
+class TestResourceManager(unittest.TestCase):
+  """Test ResourceManager module."""
+
+  def setUp(self):
+    """Test setup."""
+    config.LoadConfig()
+    self.tmp_dir = tempfile.mkdtemp(prefix='turbinia-test-tmp')
+    config.RESOURCE_FILE = os.path.join(self.tmp_dir, 'turbinia-state.json')
+
+  def tearDown(self):
+    """Tears Down temporary directory."""
+    if 'turbinia-test-tmp' in self.tmp_dir:
+      shutil.rmtree(self.tmp_dir)
+
+  def testRetrieveStateFile(self):
+    """Tests the RetrieveResourceState() method."""
+    # Test call was succesful
+    self.assertEqual(resource_manager.RetrieveResourceState(), {})
+
+    # Test file was created
+    self.assertTrue(os.path.exists(config.RESOURCE_FILE))
+
+    # Test bad resource file
+    with open(config.RESOURCE_FILE, 'w') as fh:
+      fh.write("blah")
+    fh.close()
```

Comment:
FYI, AFAIK, this shouldn't be needed as it is automatically closed outside of the `with` context manager.